certbot/src/certbot/_internal/plugins/webroot.py

```diff
@@ -118,12 +118,12 @@ def _set_webroots(self, achalls: Iterable[AnnotatedChallenge]) -> None:
             logger.info("Using the webroot path %s for all unmatched domains.",
                         webroot_path)
             for achall in achalls:
-                self.conf("map").setdefault(achall.domain, webroot_path)
+                self.conf("map").setdefault(achall.identifier.value, webroot_path)
```

Comment:
does webroot really just work with IP addresses? i wouldn't expect it to without some changes to the very strange code at https://github.com/certbot/certbot/blob/e9050f1a3c65b4e43ba802255b0789a25d79befb/certbot/src/certbot/_internal/plugins/webroot.py#L287-L327

for now i suggest we also error out in webroot in `perform` (before `_create_challenge_dirs` is called to simplify `cleanup`) and then fixup webroot and add tests for it in another PR

that sound ok to you?